src/components/views/rooms/RoomTile.js

```diff
@@ -303,6 +303,7 @@ module.exports = createReactClass({
             'mx_RoomTile_noBadges': !badges,
             'mx_RoomTile_transparent': this.props.transparent,
             'mx_RoomTile_hasSubtext': subtext && !this.props.collapsed,
+            'mx_RoomTile_isEncrypted': MatrixClientPeg.get().isRoomEncrypted(this.props.room.roomId),
```

Comment:
This probably needs to trigger a re-render when encryption is turned on in a room (although it may happen to work if the room list is updated for some other reason at that point).

Reply:
the obvious answer is to have every room tile watch all room events, filter for themselves and spot encryption events. this smacks of a performance issue, so i'm going to wander up the stack and see if i can do it above the individual tile level

Reply:
js-sdk has event emitters on the scope of a room state, so you could just bind to `room.currentState.on("RoomState.events")` so that they don't end up listening to the same events

Reply:
aha, that's useful to know. I tried something along those lines (I think I was tripped up by putting the listener on the wrong object), but it turns out there's actually something tucked away in the `room` props that lets us know if encryption was present, and that does the trick

Reply:
you *should* use isRoomEncrypted as it has some additional logic to prevent a redaction from applying, the room prop also doesn't get updated when the depths of the room state change so this would not cause a re-render if the room went from unencrypted to encrypted